### PR TITLE
bump cni

### LINKED ISSUE #31630: Don't allocate subnet broadcast address
I have a container that got the broadcast address of podCIDR subnet. 
My CNI conf is:
```
{
  "cniVersion": "0.1.0",
  "name": "kubenet",
  "type": "bridge",
  "bridge": "cbr0",
  "mtu": 1460,
  "addIf": "eth0",
  "isGateway": true,
  "ipMasq": false,
  "hairpin": "false",
  "ipam": {
    "type": "host-local",
    "subnet": "10.180.2.0/24",
    "gateway": "10.180.2.1",
    "routes": [
      { "dst": "0.0.0.0/0" }
    ]
  }
}
```

And the container in question has `10.180.2.255/24` which is the broadcast address for the subnet. So it can't figure out the MAC of cbr0 because all frames are getting sent to `ff:ff:ff:ff:ff:ff` (or at least, that's what `arp -n` seems to indicate). 

Eg: tcpdump alonside a wget on google.com
```
beeps@e2e-test-beeps-minion-group-4737:~$ sudo tcpdump -e -i cbr0 -n host 10.180.2.255
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on cbr0, link-type EN10MB (Ethernet), capture size 65535 bytes
19:09:25.644425 a2:c7:88:bc:5c:e0 > ff:ff:ff:ff:ff:ff, ethertype ARP (0x0806), length 42: Request who-has 10.180.2.1 tell 10.180.2.255, length 28
19:09:26.640693 a2:c7:88:bc:5c:e0 > ff:ff:ff:ff:ff:ff, ethertype ARP (0x0806), length 42: Request who-has 10.180.2.1 tell 10.180.2.255, length 28
19:09:27.640721 a2:c7:88:bc:5c:e0 > ff:ff:ff:ff:ff:ff, ethertype ARP (0x0806), length 42: Request who-has 10.180.2.1 tell 10.180.2.255, length 28
```

vs a similar normal container
```
beeps@e2e-test-beeps-minion-group-4737:~$ sudo tcpdump -e -i cbr0 -n host 10.180.2.171
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on cbr0, link-type EN10MB (Ethernet), capture size 65535 bytes
19:09:44.965805 c2:9d:c9:0d:e5:bc > 22:cf:99:33:19:ff, ethertype IPv4 (0x0800), length 83: 10.180.2.171.36148 > 169.254.169.254.53: 3769+ PTR? 1.2.180.10.in-addr.arpa. (41)
19:09:44.969367 22:cf:99:33:19:ff > c2:9d:c9:0d:e5:bc, ethertype IPv4 (0x0800), length 171: 169.254.169.254.53 > 10.180.2.171.36148: 3769 NXDomain 0/1/0 (129)
```

@kubernetes/sig-network

### PR DESCRIPTION
**Pull request by freehan**
closes: #31630 #31724

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31801)

### CODE REVIEW

File: cluster/saltbase/salt/cni/init.sls [hunk 1]
```diff
@@ -19,9 +19,9 @@ cni-tar:
     - user: root
     - name: /opt/cni
     - makedirs: True
-    - source: https://storage.googleapis.com/kubernetes-release/network-plugins/cni-9d5e6e60e79491207834ae8439e80c943db65a69.tar.gz
+    - source: https://storage.googleapis.com/kubernetes-release/network-plugins/cni-a1802e47123688f6e84d8b297ce5a95ce724ac27.tar.gz
     - tar_options: v
-    - source_hash: md5=1ac2f8c72244f0188b8ba3d6052f9fc1
+    - source_hash: md5=6a8c115d040e6a24695bdcc9d8f6622d
```
Comment (bprashanth) [inline cluster/saltbase/salt/cni/init.sls]: If we're not going to fork into third_party and the upstream pr isn't going to make it in time, please add a RELEASES file to the same bucket with exactly what this release contains. Also add a comment here and in the gci script that this is not just the stock CNI binary, with a pointer to the RELEASES file.

**Comment by bprashanth** [comment]
@kubernetes/sig-network would be good to get https://github.com/containernetworking/cni/pull/292 merged. The alternatives are to fork the CNI plugins into kubernetes/third_party, or just push the updated binary with the cherrypicked pr, neither of which are very appealing.

**Comment by bprashanth** [comment]
I can't debug the e2e failure we didn't collect master logs and cluster failed to come up, @freehan suggest  just re-upping which will force retest

**Comment by bprashanth** [comment]
@freehan what's the impact of getting this in as is? we sometimes wrongfully don't allocate one ip, correct? If that's the case I don't think it matters too much, our default subnets are 2x max pods we allow on a node anyway. How about we just get this in so we get test runtime over the weekend? If the follow up fix makes it in time, we take it.

**Comment by freehan** [comment]
Yeah. I thought so too. Essentially it is an one line fix. Having this in could help us clear up some flakes and focus on others.

**Comment by bprashanth** [comment]
Alright lets try to do that. Still suggest maintaining some checked in record of exactly what we have in this build that isn't in upstream in the form of a CHANGELOG file but you can send a follow up pr for that if this goes all green and you want to avoid the e2e turn around time. 

You can even create a pkg/build directory just like we do for kube-dns

**Comment by bprashanth** [comment]
I think your hash is wrong
```
$ wget -O - https://storage.googleapis.com/kubernetes-release/network-plugins/cni-a1802e47123688f6e84d8b297ce5a95ce724ac27.tar.gz | tee cni-a1802e47123688f6e84d8b297ce5a95ce724ac27.tar.gz | md5sum
--2016-09-02 15:31:37--  https://storage.googleapis.com/kubernetes-release/network-plugins/cni-a1802e47123688f6e84d8b297ce5a95ce724ac27.tar.gz
Resolving storage.googleapis.com (storage.googleapis.com)... 2607:f8b0:4005:804::2010, 216.58.192.16
Connecting to storage.googleapis.com (storage.googleapis.com)|2607:f8b0:4005:804::2010|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 16246929 (15M) [application/gzip]
Saving to: ‘STDOUT’

100%[===================================================================================================================================================================================================================================================================================================================================>] 16,246,929  23.2MB/s   in 0.7s   

2016-09-02 15:31:38 (23.2 MB/s) - written to stdout [16246929/16246929]

8643b8577601303a90e49f05e6931ad5  -
```

**Comment by bprashanth** [comment]
what is this, why is it failing?? 
did you actually try a kube-up :)

**Comment by freehan** [comment]
Redo it from scratch and push the new binary with https://github.com/containernetworking/cni/pull/294

**Comment by bprashanth** [comment]
Is this just HEAD upstream or some release with a cherrypick? 
If it's the former, suggest stress testing locally to make sure we didn't pick up an stray changes this close to the release. 
LGTM pending tests

**Comment by freehan** [comment]
It is upstream HEAD. Will try to check the diffs in related components.

**Comment by foxish** [comment]
Already in the 1.4 branch, removing cherrypick candidate label. 
@bprashanth @freehan

**Pull request merged**